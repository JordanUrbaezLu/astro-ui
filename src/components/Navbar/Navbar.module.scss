@use "../../utility/globals";

.navbarContainer {
  display: flex;
  flex-direction: row;
  position: relative;
  justify-content: flex-end;
  font-size: 16px;
  min-width: 100vw;
  font-family: Verdana, Geneva, Tahoma, sans-serif;

  &.primary {
    background-color: globals.$primary-color-pseudo-class;
  }
  
  &.secondary {
      background-color: globals.$secondary-color-pseudo-class;
  }
}

.listContainer {
  display: flex;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.navbarLogoContainer {
  background-color: globals.$primary-color-pseudo-class;
  position: absolute;
  left: 0;
  max-height: 58.5px;
}

.navbarItemContainer {
  display: flex;
}

.navbarLogo {
  background-color: globals.$primary-color-pseudo-class;
  max-height: 58.5px;
  width: 100%;

  &.primary {
    background-color: globals.$primary-color-pseudo-class;
  }
  
  &.secondary {
      background-color: globals.$secondary-color-pseudo-class;
  }
}

