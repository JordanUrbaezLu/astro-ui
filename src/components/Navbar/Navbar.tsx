import classNames from "classnames";
import * as React from "react";
import styles from "./Navbar.module.scss";
import itemStyles from "./NavbarItem.module.scss";

export interface NavbarLogoProps {
  /**
   * The link for the Navbar's logo
   */
  link: string;
  /**
   * The url for the Navbar's logo
   */
  url: string;
}

export type NavbarVariant = "primary" | "secondary";

export interface NavbarProps {
  /**
   * The content for the Navbar
   */
  children?: React.ReactNode;
  /**
   * The props for the Navbar's logo
   */
  logoProps?: NavbarLogoProps;
  /**
   * The variant of the Navbar
   *
   * @default "primary"
   */
  variant?: NavbarVariant;
}

export const Navbar: React.FC<NavbarProps> = (props) => {
  const { children, logoProps, variant = "primary" } = props;

  const navbarContainer = classNames(
    styles.navbarContainer,
    variant === "primary" ? styles.primary : styles.secondary
  );

  const NavbarLogo = classNames(
    styles.NavbarLogo,
    variant === "primary" ? styles.primary : styles.secondary
  );

  return (
    <nav className={navbarContainer} role="navigation">
      {logoProps && (
        <a
          className={styles.NavbarLogoContainer}
          href={logoProps?.link ? logoProps.link : "#"}
        >
          <img className={NavbarLogo} src={logoProps?.url} alt="logo" />
        </a>
      )}
      <ul className={styles.listContainer}>
        {React.Children.map(children, (child) => {
          const navbarItem = child as React.ReactElement;

          return (
            <li
              className={classNames(
                variant === "primary"
                  ? itemStyles.primary
                  : itemStyles.secondary,
                styles.NavbarItemContainer
              )}
            >
              {navbarItem}
            </li>
          );
        })}
      </ul>
    </nav>
  );
};
