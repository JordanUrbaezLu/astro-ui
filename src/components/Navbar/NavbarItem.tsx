import * as React from "react";
import styles from "./NavbarItem.module.scss";

export interface NavbarItemProps {
  /**
   * The content for the NavbarItem
   */
  children: React.ReactNode;
  /**
   * The url link for the NavbarItem
   */
  link: string;
}

export const NavbarItem: React.FC<NavbarItemProps> = (props) => {
  const { children, link } = props;
  return (
    <a className={styles.navbarItemLinkContainer} href={link}>
      {children}
    </a>
  );
};
