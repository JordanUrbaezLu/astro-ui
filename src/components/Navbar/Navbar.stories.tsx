import { Navbar } from "./Navbar";
import { NavbarItem } from "./NavbarItem";

export default {
  component: Navbar,
  title: "Components/Navbar",
};

export const Default = () => {
  return (
    <Navbar>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
    </Navbar>
  );
};

Default.parameters = {
  layout: "fullscreen",
};

export const Logo = () => {
  return (
    <Navbar
      logoProps={{
        link: "https://github.com/JordanUrbaezLu/void-ui",
        url: "https://www.pngkey.com/png/full/141-1415366_arctic-cat-logo-github-png.png",
      }}
    >
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
    </Navbar>
  );
};

Logo.parameters = {
  layout: "fullscreen",
};

export const Secondary = () => {
  return (
    <Navbar
      logoProps={{
        link: "https://github.com/JordanUrbaezLu/void-ui",
        url: "https://www.pngkey.com/png/full/141-1415366_arctic-cat-logo-github-png.png",
      }}
      variant="secondary"
    >
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
      <NavbarItem link="https://github.com/JordanUrbaezLu/void-ui">
        Github
      </NavbarItem>
    </Navbar>
  );
};

Secondary.parameters = {
  layout: "fullscreen",
};
