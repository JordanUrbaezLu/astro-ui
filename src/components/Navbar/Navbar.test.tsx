import { screen, render } from "@testing-library/react";
import { axe } from "jest-axe";
import { Navbar, NavbarVariant } from "./Navbar";
import { NavbarItem } from "./NavbarItem";

describe("Calendar", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <Navbar>
        <NavbarItem link="https://github.com/JordanUrbaezLu/-ui-library">
          Github
        </NavbarItem>
      </Navbar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(
      <Navbar>
        <NavbarItem link="https://github.com/JordanUrbaezLu/-ui-library">
          Github
        </NavbarItem>
      </Navbar>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render children correctly.", () => {
    render(
      <Navbar>
        <NavbarItem link="https://github.com/JordanUrbaezLu/-ui-library">
          Github
        </NavbarItem>
      </Navbar>
    );
    expect(screen.getByText("Github")).toBeInTheDocument();
  });
  test("Should render logo props correctly.", () => {
    render(
      <Navbar
        logoProps={{
          link: "https://github.com/JordanUrbaezLu/-ui-library",
          url: "https://www.pngkey.com/png/full/141-1415366_arctic-cat-logo-github-png.png",
        }}
      >
        <NavbarItem link="https://github.com/JordanUrbaezLu/-ui-library">
          Github
        </NavbarItem>
      </Navbar>
    );
    expect(
      document.getElementsByClassName("NavbarLogoContainer")[0]
    ).toBeInTheDocument();
  });
  test.each<NavbarVariant>(["primary", "secondary"])(
    "Should render variant correctly.",
    (variant) => {
      render(
        <Navbar variant={variant}>
          <NavbarItem link="https://github.com/JordanUrbaezLu/-ui-library">
            Github
          </NavbarItem>
        </Navbar>
      );
      expect(screen.getByRole("navigation")).toHaveClass(variant);
    }
  );
});
