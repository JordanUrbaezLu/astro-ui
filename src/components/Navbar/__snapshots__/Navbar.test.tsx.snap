// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Should render component correctly. 1`] = `
<nav
  class="navbarContainer primary"
  role="navigation"
>
  <ul
    class="listContainer"
  >
    <li
      class="primary NavbarItemContainer"
    >
      <a
        class="navbarItemLinkContainer"
        href="https://github.com/JordanUrbaezLu/-ui-library"
      >
        Github
      </a>
    </li>
  </ul>
</nav>
`;
